### PR TITLE
Added WgA limits features user can set

### PR DESCRIPTION
**Pull request by SixZeng**
### Briefly, what does this PR introduce?
It allows users to set a limit for the center-of-mass energy of the virtual photon and the arbitrary nucleus, W_gA. 

### What kind of change does this PR introduce?
- [ ] Bug fix 
- [x] New feature 
- [x] Documentation update
- [ ] Other: __

### Please check if this PR fulfills the following:
- [x] Tests for the changes have been added
- [x] Documentation has been added / updated
- [x] Changes have been communicated to collaborators

### Does this PR introduce breaking changes? What changes might users need to make to their code?
No breaking changes. 
Users may set up limits for the WgA, W_gA_MIN and W_gA_MAX, in the slight.in file. If not needed, please comment out the W_GA_MAX and W_GA_MIN.  
### Does this PR change default behavior?
No.

### CODE REVIEW

**Comment by zsweger** [comment]
@wdconinc Do you recognize this pre-commit.ci "error during ci config" message? I've never seen this on eSTARlight before and I didn't think we had CI set up for this repo. Any thoughts would be helpful

**Comment by wdconinc** [comment]
I think one of the other admins of the eic workspace clicked a wrong button and enabled pre-commit.ci on all repositories. I'll disable it again and only run on select repositories.

File: src/photonNucleusCrossSection.cpp [hunk 1]
```diff
@@ -909,7 +911,8 @@ photonNucleusCrossSection::sigmagp(const double Wgp)
 	// Function for the gamma-proton --> VectorMeson
 	// cross section. Wgp is the gamma-proton CM energy.
 	// Unit for cross section: fm**2
-  
+	//If W_gp is not in the allowed region, i.e. W_GP_MIN < W_gp < W_GP_MAX, do not sample.
+  if(Wgp < _minW_GP || Wgp > _maxW_GP) return 0;
```
Comment (wdconinc) [inline src/photonNucleusCrossSection.cpp]: indent

**Review by wdconinc — COMMENTED** [review]
I can't really comment on the physics for this review :-D

**Comment by zsweger** [comment]
It worked! Thank you!

**Comment by wdconinc** [comment]
Nice, the last commit didn't trigger the pre-commit check.

**Pull request merged**